src/application-base/ApplicationBase.jsx

```diff
@@ -32,97 +33,71 @@ const propTypes = {
   children: PropTypes.node.isRequired,
   /**
    * The locale name to be used to load translated messages.
-   * If the `locale` prop is not provided, the preferred language from the browser will be used.
+   * If the `locale` prop is not provided, the preferred language from the
+   * browser will be used.
    */
   locale: PropTypes.string,
   /**
-   * The name of the theme to apply to the application using terra-theme-provider.
+   * The name of the theme to apply to the application.
    */
   themeName: PropTypes.string,
-  /**
-   * By default, NavigationPrompts rendered within ApplicationBase will cause the user to be prompted during
-   * the window's beforeUnload event. If `unloadPromptIsDisabled` is provided, the user will **not** be prompted
-   * before continuing with the unload event, even if NavigationPrompts are present.
-   */
-  unloadPromptIsDisabled: PropTypes.bool,
 };
 
 const ApplicationBase = ({
-  locale, themeName, children, unloadPromptIsDisabled,
+  locale, themeName, children,
 }) => {
-  const registeredPromptsRef = useRef();
   const [messages, setMessages] = useState();
 
-  const { localeOverride } = useTestOverrides(); // Allows us to test deployed applications in different locales.
+  // Allows us to test deployed applications in different locales.
+  const { localeOverride } = useTestOverrides();
 
   const finalLocale = localeOverride || locale || browserLocale;
 
   useEffect(() => {
     if (finalLocale !== undefined) {
-      i18nLoader(finalLocale).then(translationsModule => setMessages(translationsModule.default)).catch((error) => {
-        // eslint-disable-next-line no-console
-        console.error(error);
-        throw error;
+      i18nLoader(finalLocale).then(translationsModule => {
+        setMessages(translationsModule.default);
+      }).catch((error) => {
+        logger.error(error);
```

Comment:
I removed the thrown error here. If locales cannot be loaded, we will log and continue rendering null. Throwing an error seemed like overkill (and it was hard to test).